cfme/tests/infrastructure/test_webmks_console.py

```diff
@@ -0,0 +1,178 @@
+
+pytest_generate_tests = testgen.generate(
+    gen_func=testgen.providers,
+    filters=[ProviderFilter(classes=[InfraProvider], required_flags=['webmks_console'])],
+    scope='module'
+)
+
+
+@pytest.fixture(scope="function")
+def vm_obj(request, provider, setup_provider, console_template, vm_name):
+    """
+    Create a VM on the provider with the given template, and return the vm_obj.
+
+    Also, it will remove VM from provider using nested function _delete_vm
+    after the test is completed.
+
+    """
+    vm_obj = VM.factory(vm_name, provider, template_name=console_template)
+
+    @request.addfinalizer
+    def _delete_vm():
+        try:
+            vm_obj.delete_from_provider()
+        except Exception:
+            logger.warning("Failed to delete vm `{}`.".format(vm_obj.name))
+
+    vm_obj.create_on_provider(timeout=2400, find_in_cfme=True, allow_skip="default")
+    return vm_obj
+
+
+@pytest.fixture(scope="module")
+def configure_vmware_console_for_test(appliance, provider):
+    """Configure VMware Console to use VNC which is what is required for the HTML5 console."""
+    navigate_to(appliance.server, 'Server')
+
+    settings_pg = VMwareConsoleSupport(
+        appliance=appliance,
+        console_type='VMware WebMKS',
+    )
+    settings_pg.update()
+
+
+@pytest.fixture(scope="session")
+def configure_websocket():
+    """
+    Enable websocket role if it is disabled.
+
+    Currently the fixture cfme/fixtures/base.py,
+    disables the websocket role to avoid intrusive popups.
+    """
+    roles = configuration.get_server_roles()
+    if 'websocket' in roles and not roles['websocket']:
+        logger.info('Enabling the websocket role to allow console connections')
+        roles['websocket'] = True
+        configuration.set_server_roles(**roles)
+        yield
+    roles['websocket'] = False
+    logger.info('Disabling the websocket role to avoid intrusive popups')
+    configuration.set_server_roles(**roles)
+
+
+@pytest.mark.uncollectif(lambda: version.current_version() < '5.8', reason='Only valid for >= 5.8')
+def test_html5_vm_console(appliance, provider, vm_obj, configure_websocket,
+        configure_vmware_console_for_test):
+    """
+    Test the HTML5 console support for a particular provider.
+
+    The supported providers are:
+
+        VMware
+        Openstack
+        RHV
+
+    For a given provider, and a given VM, the console will be opened, and then:
+
+        - The console's status will be checked.
+        - A command that creates a file will be sent through the console.
+        - Using ssh we will check that the command worked (i.e. that the file
+          was created.
+    """
+    console_vm_username = credentials[provider.data.templates.get('console_template')
+                            ['creds']].get('username')
+    console_vm_password = credentials[provider.data.templates.get('console_template')
+                            ['creds']].get('password')
```

Comment:
Is the 'console_template' key guaranteed to be always there? This will fail with TypeError if not.